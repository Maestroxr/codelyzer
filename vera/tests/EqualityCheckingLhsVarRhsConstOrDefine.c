
#define CONSTANT_DEFINE 0
int main()
{
	int returnValue = 0;
	
	const int CONSTANT_INT = 0;
	int x = 0;
	
	/*OK*/
	if (CONSTANT_INT == x) 
	{
		returnValue = 1;
	}
	
	if (CONSTANT_DEFINE == x) 
	{
		returnValue = 1;
	}
	/*--------*/
	
	
	/*PROBLEM*/
	if (x == CONSTANT_INT) 
	{
		returnValue = -1;
	}

	if (x == CONSTANT_DEFINE) 
	{
		returnValue = -1;
	}
	/*--------*/
	
	
	return returnValue;
}

