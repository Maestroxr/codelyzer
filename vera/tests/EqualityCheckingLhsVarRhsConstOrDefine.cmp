C:\Work\AutoAssessment\codelyzer\vera\tests\EqualityCheckingLhsVarRhsConstOrDefine.c:34: Equality checking 'problem1 == CONSTANT_INT' with left-hand side argument as variable and right-hand side as const or #define.C:\Work\AutoAssessment\codelyzer\vera\tests\EqualityCheckingLhsVarRhsConstOrDefine.c:39: Equality checking 'problem2 == CONSTANT_DEFINE' with left-hand side argument as variable and right-hand side as const or #define.