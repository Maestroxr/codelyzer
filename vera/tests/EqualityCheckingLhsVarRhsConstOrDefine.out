C:\Work\AutoAssessment\codelyzer\vera\tests\EqualityCheckingLhsVarRhsConstOrDefine.c:24: Equality checking 'x == CONSTANT_INT' with left-hand side argument as variable and right-hand side as const or #define.
C:\Work\AutoAssessment\codelyzer\vera\tests\EqualityCheckingLhsVarRhsConstOrDefine.c:29: Equality checking 'x == CONSTANT_DEFINE' with left-hand side argument as variable and right-hand side as const or #define.
